Paws4Me/Paws4Me/Controllers/Logic/ViewController.swift

```diff
@@ -8,109 +8,243 @@
-        if let destination = segue.destination as? PetSingleDetailsViewController {
-            guard let rowIndex = petTable.indexPathForSelectedRow?.row else { return }
-            guard let pageItem = adoptPetObject?.page?[rowIndex] else { return }
-            if let pagePetName = pageItem.name {
-                destination.namePet = pagePetName
-            }
-            if let pagePetAge = pageItem.age {
-                destination.agePet = pagePetAge
-            }
-            if let pageGender = pageItem.sex {
-                destination.genderPet = pageGender
-            }
-            if let pagePetImage = pageItem.animalImage {
-                destination.imgPet = pagePetImage
-            }
-            if let pagePetBreed = pageItem.animalSpeciesBreed?.petBreedName {
-                destination.breedPet = pagePetBreed
-            }
-        }
-    }
-    func searchBar(_ searchBar: UISearchBar, textDidChange searchText: String) {
-        self.filteredPetObject = nil
-        if searchText == "" {
-            filteredPetObject = adoptPetObject
-        } else {
-            guard let petObject = adoptPetObject else {
-                return
-            }
-            if  searchText.lowercased().contains("Female") {
-                filteredPetObject = petObject
-            }
         }
-        self.petTable.reloadData()
-    }
-}
-extension UIViewController {
-    func displayAlert(title: String, message: String, buttonTitle: String) {
-        let alert = UIAlertController(title: title, message: message, preferredStyle: UIAlertController.Style.alert)
-        alert.addAction(UIAlertAction(title: buttonTitle, style: UIAlertAction.Style.default, handler: nil))
-        self.present(alert, animated: true, completion: nil)
     }
 }
+
+   extension ViewController: UITableViewDelegate, UITableViewDataSource {
+       func setUpSearchbar() {
+           searchBarController = UISearchBar(frame: CGRect(x: 0, y: 0, width: self.petTable.bounds.width, height: 65))
+           searchBarController.showsScopeBar = true
+           searchBarController.scopeButtonTitles = ["Male", "Female"]
+           searchBarController.selectedScopeButtonIndex = 0
+           searchBarController.delegate = self
+           self.petTable.tableHeaderView = searchBarController
+       }
+
+       // MARK: - Tableview
+       func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+           if let pageCount = filteredPetObject?.page?.count {
+               return pageCount
+           } else {
+               return 0
+           }
+       }
+       func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+           guard let cell = tableView.dequeueReusableCell(withIdentifier: "petAdoptCell") as? AnimalTableViewCell else {
+               return UITableViewCell()
+           }
+           let adoptablepet = filteredPetObject
```

Comment:
Why do you need this variable if you already have filteredPetObject available to use?